sycl/doc/extensions/proposed/sycl_ext_intel_cache_controls.asciidoc

```diff
@@ -92,178 +92,217 @@ supports.
-to the processing unit, cache level 1 indicates the next furthest cache
-level, etc. It is legal to specify a cache level that does not exist on
-the target device, but the property will be ignored in this case.
-
-Note that a property specifies the cache behavior only for the indicated
-cache level. In order to specify the behavior for multiple cache levels, 
-multiple properties should be specified.
-
-It is legal to specify several different cache control properties in the
-same `annotated_ptr`. However, all instances of `cache_control_read_type` must
-have different cache levels and all instances of `cache_control_write_type`
-must have difference cache levels.
-
-The cache control properties are divided into two categories: those that
-are hints and those that are assertions by the application.
-
-==== Cache control hints
-These properties are hints requesting specific cache behavior when
-loading or storing to memory through the `annotated_ptr`. These properties can
-affect the performance of device code, but they do not change the semantics.
-
 --
-[options="header", cols="2,1"]
+[options="header", cols="3,2"]
 |====
 | Property | Description
 a|
 [source]
 ----
-cache_control_read<cache_control_read_type::cached, L>
+read_hint<cache_control<cache_mode::uncached, Ls...>>
 ----
 |
 This property requests that loads from memory through the `annotated_ptr`
-may cache the data at level `L` in the memory hierarchy.
+should not cache the data at levels `Ls` in the memory hierarchy.
 a|
 [source]
 ----
-cache_control_read<cache_control_read_type::uncached, L>
+read_hint<cache_control<cache_mode::cached, Ls...>>
 ----
 |
 This property requests that loads from memory through the `annotated_ptr`
-should not cache the data at level `L` in the memory hierarchy.
+may cache the data at levels `Ls` in the memory hierarchy.
 a|
 [source]
 ----
-cache_control_read<cache_control_read_type::streaming, L>
+read_hint<cache_control<cache_mode::streaming, Ls...>>
 ----
 |
 This property requests that loads from memory through the `annotated_ptr`
-should cache the data at cache level `L`. The eviction policy is to give
+should cache the data at cache levels `Ls`. The eviction policy is to give
 lower priority to data cached using this property versus the `cached`
 property.
+|====
+--
+
+==== Write hint property
+This property is a hint requesting specific cache behavior when
+storing to memory through an `annotated_ptr`.
+This property can affect the performance of device code, but it
+does not change the semantics.
+
+```c++
+namespace sycl::ext {
+namespace intel::experimental {
+
+struct write_hint_key {
+  template <typename... Cs>
+  using value_t = property_value<write_hint_key, Cs...>;
+};
+
+template <typename... Cs>
+inline constexpr write_hint_key::value_t<Cs...> write_hint;
+
+} // namespace intel::experimental
+
+namespace oneapi {
+namespace experimental {
+
+using write_hint_key = intel::experimental::write_hint_key;
```

Comment:
Same namespace pollution problem here.  Also for `read_assertion_key` below.

Reply:
Removed the using statements that caused namespace pollution in doc and implementation.